### PR TITLE
feat: add basic shellchecking

### PR DESCRIPTION
**Pull request by MitchellPaff**
Pushing through commits from other branch in piecemeal fashion.

### CODE REVIEW

File: scripts/linz-bde-schema-load.bash [hunk 1]
```diff
@@ -84,7 +84,7 @@ dbpatch-loader - fake 2>&1 | grep -q "database.*does not exist" &&
     DBPATCH_SUPPORTS_STDOUT=no ||
     DBPATCH_SUPPORTS_STDOUT=yes
 
-if test $PGDATABASE = "-" -a $DBPATCH_SUPPORTS_STDOUT != yes; then
+if test "$PGDATABASE" = "-" -a $DBPATCH_SUPPORTS_STDOUT != yes; then
```
Comment (l0b0) [inline scripts/linz-bde-schema-load.bash:87]: `$DBPATCH_SUPPORTS_STDOUT` should be quoted here.

File: scripts/linz-bde-schema-load.bash [hunk 2]
```diff
@@ -98,7 +98,7 @@ fi
 if test $DBPATCH_SUPPORTS_STDOUT != yes; then
     echo "WARNING: dbpatch-loader does not support stdout mode, working in non-transactional mode" >&2
     echo "HINT: install dbpatch 1.4.0 or higher to fix this." >&2
-    ${DBPATCH_LOADER} ${DBPATCH_OPTS} ${PGDATABASE} _patches || exit 1
+    ${DBPATCH_LOADER} ${DBPATCH_OPTS} "${PGDATABASE}" _patches || exit 1
```
Comment (l0b0) [inline scripts/linz-bde-schema-load.bash:101]: All of these variables should be quoted. ShellCheck unfortunately doesn't suggest quoting variables which are currently set to a static string with no special characters.

File: scripts/linz-bde-schema-load.bash [hunk 3]
```diff
@@ -147,26 +147,26 @@ if test "${ADD_REVISIONS}" = "yes" -a $TABLEVERSION_SUPPORTS_STDOUT = yes; then
...
 
 cat << EOF
 CREATE EXTENSION IF NOT EXISTS postgis SCHEMA public;
 EOF
 
-for file in ${SCRIPTSDIR}/*.sql; do
+for file in "${SCRIPTSDIR}"/*.sql; do
     if test ${SKIP_INDEXES} = 'yes' &&
-       `basename $file .sql` = '04-bde_schema_index';
+       [ "$(basename "$file" .sql)" = '04-bde_schema_index' ];
```
Comment (l0b0) [inline scripts/linz-bde-schema-load.bash]: Same as [elsewhere](https://github.com/linz/linz-bde-schema/pull/232#discussion_r796113147), this is probably a bug.

File: scripts/linz-bde-schema-load.bash [hunk 4]
```diff
@@ -138,34 +139,34 @@ fi
...
     if [[ "$SKIP_INDEXES" = 'yes' ]] && [[ "$(basename "$file" .sql)" = '04-bde_schema_index' ]]
     then
         continue
     fi
-    cat ${file} || rollback
+    cat "${file}" || rollback
 done
 
 if test "${ADD_REVISIONS}" = "yes"; then
-    file=${SCRIPTSDIR}/versioning/01-version_tables.sql
```
Comment (MitchellPaff) [inline scripts/linz-bde-schema-load.bash:167]: Is this correct?

Comment (l0b0) [reply scripts/linz-bde-schema-load.bash:167]: Yep, your change is correct.

**Review by l0b0 — APPROVED** [review]

**Pull request merged**